docs/en/operations/settings/settings.md

```diff
@@ -404,6 +404,32 @@ Possible values:
 
 Default value: 0.
 
+## any_join_distinct_right_table_keys {#any_join_distinct_right_table_keys}
+
+Enables or disables specific ClickHouse server behavior in `ANY INNER|LEFT|RIGHT| JOIN` operations.
+
+When the behavior enabled:
+
+- Results of `t1 ANY LEFT JOIN t2` and `t2 ANY RIGHT JOIN t1` operations are not equal because ClickHouse uses the logic with many-to-one left-to-right table keys mapping.
+- Results of `ANY INNER JOIN` operations contain all rows from the left table like the `SEMI LEFT JOIN` operations do.
+
+When the behavior disabled:
+
+- Results of `t1 ANY LEFT JOIN t2` and `t2 ANY RIGHT JOIN t1` operations are equal because ClickHouse uses the logic which provides one-to-many keys mapping in `ANY RIGHT JOIN` operations.
+- Results of `ANY INNER JOIN` operations contain one row per key from both left and right tables.
+
+Possible values:
+
+- 1 — ClickHouse uses the logic with many-to-one left-to-right table keys mapping in `ANY JOIN` operations.
+- 0 — ClickHouse uses the logic with one-to-many keys in `ANY RIGHT JOIN` operations.
```

Comment:
Very strange choice. It covers only 4/6 cases. 0 with LEFT and INNER JOINS are not covered.